google-cloud-examples/src/main/java/com/google/cloud/examples/bigtable/TableAdminExample.java

```diff
@@ -0,0 +1,373 @@
+
+    // Creates column family with given GC rule.
+    try {
+      ModifyColumnFamiliesRequest columnFamiliesRequest2 =
+          ModifyColumnFamiliesRequest.of(tableId).addFamily(COLUMN_FAMILY_2, versionRule1);
+      adminClient.modifyFamilies(columnFamiliesRequest2);
+      System.out.println("Created column family: " + COLUMN_FAMILY_2);
+    } catch (AlreadyExistsException e) {
+      System.err.println(
+          "Failed to create column family with rule, already exists: " + e.getMessage());
+    }
+    // [END bigtable_create_family_gc_max_versions]
+  }
+
+  /** Demonstrates how to create a new instance of the UnionRule. */
+  public void unionRule() {
+    System.out.printf("%nCreating column family %s with union GC rule%n", COLUMN_FAMILY_3);
+    // [START bigtable_create_family_gc_union]
+    // Creates a column family with GC policy to drop data that matches at least one condition.
+
+    // Defines a GC rule to drop cells older than 5 days OR not the most recent version.
+    GCRules.DurationRule maxAgeRule2 = GCRules.GCRULES.maxAge(5, TimeUnit.DAYS);
+    GCRules.VersionRule versionRule2 = GCRules.GCRULES.maxVersions(1);
+    // Add rules to union rule list
+    GCRules.UnionRule unionRule1 = GCRules.GCRULES.union().rule(maxAgeRule2).rule(versionRule2);
+
+    // Creates column family with given GC rule.
+    try {
+      ModifyColumnFamiliesRequest columnFamiliesRequest3 =
+          ModifyColumnFamiliesRequest.of(tableId).addFamily(COLUMN_FAMILY_3, unionRule1);
+      adminClient.modifyFamilies(columnFamiliesRequest3);
+      System.out.println("Created column family: " + COLUMN_FAMILY_3);
+    } catch (AlreadyExistsException e) {
+      System.err.println(
+          "Failed to create column family with rule, already exists: " + e.getMessage());
+    }
+    // [END bigtable_create_family_gc_union]
+  }
+
+  /** Demonstrates how to create a new instance of the IntersectionRule. */
+  public void intersectionRule() {
+    System.out.printf("%nCreating column family %s with intersection GC rule%n", COLUMN_FAMILY_4);
+    // [START bigtable_create_family_gc_intersection]
+    // Creates a column family with GC policy to drop data that matches all conditions.
+
+    // Defines a GC rule to drop cells older than 5 days AND older than the most recent 2 versions.
+    GCRules.DurationRule maxAgeRule3 = GCRules.GCRULES.maxAge(5, TimeUnit.DAYS);
+    GCRules.VersionRule versionRule3 = GCRules.GCRULES.maxVersions(2);
+    GCRules.IntersectionRule intersectionRule1 =
+        GCRules.GCRULES.intersection().rule(maxAgeRule3).rule(versionRule3);
+
+    // Creates column family with given GC rule.
+    try {
+      ModifyColumnFamiliesRequest columnFamiliesRequest4 =
+          ModifyColumnFamiliesRequest.of(tableId).addFamily(COLUMN_FAMILY_4, intersectionRule1);
+      adminClient.modifyFamilies(columnFamiliesRequest4);
+      System.out.println("Created column family: " + COLUMN_FAMILY_4);
+    } catch (AlreadyExistsException e) {
```

Comment:
A modify shouldnt throw an AlreadyExistsException

Reply:
`ModifyColumnFamiliesRequest` is also used for **adding** a new columnFamily after the table is already created, not just modifying. So, if someone tries to add a columnFamily with the same name as an existing one, `AlreadyExistsException` will be thrown.